blockchain/tx_pool.go

```diff
@@ -769,7 +769,7 @@ func (pool *TxPool) validateTx(tx *types.Transaction) error {
 		// balance check for fee-delegated tx
 		gasFeePayer, err = tx.ValidateFeePayer(pool.signer, pool.currentState, pool.currentBlockNumber)
 		if err != nil {
-			return ErrInvalidFeePayer
+			return err
```

Comment:
This change provides more detailed error messages to users, but it may confuse them. A user will receive the same error message for an invalid sender's signature and an invalid feePayer's signature. For example, "invalid chain id for signer" message can be returned for both invalid sender and feePayer.

Reply:
If you want to get a more detailed error message, we may add an additional messages such as "from fee payer" at the end of `err`.

Reply:
To preserve original error messges, we could do:
```go
// error.go
func ErrInvalidSender(reason error) error {
    return fmt.Errorf("invalid sender: %s", reason)
}
func ErrInvalidFeePayer(reason error) error {
    return fmt.Errorf("invalid fee payer: %s", reason)
}
```
And use ErrInvalidSender at line 756, ErrInvalidFeePayer at line 772.

Reply:
Is it only for fee delegated transactions? I can't find the invalid chain id error in `tx.ValidateFeePayer()`

Reply:
I understood it. Thanks for the suggestion. I think @blukat29's suggestion is better in this case.